### PR TITLE
Add support for using server-icon.png as favIcon

### PR DESCRIPTION
**Pull request by markmetcalfe**
Currently the only way to override the default server icon is to specify a large base64 string in the settings config file. This is annoying as it requires you to manually convert the image you want into base64, and then place the string into the file which is bad for performance - I noticed VS Code was slower when opening the file.

With this change, you can simply just add
```
useServerIcon: true
```
into the settings yml, and it will load the `server-icon.png` file in the minecraft server directory, which is much more convenient.

### CODE REVIEW

File: src/sleepingHelper.ts [hunk 1]
```diff
@@ -18,6 +20,20 @@ export const isPortTaken = (port: number) => new Promise<boolean>((resolve) => {
...
...nY1PTAqvr6wpF5AKbC2NJD1kpVANHH7BerG5vo549/j47s3Uu9Fy9qA1Aep2Vd8wFkRoiBcLZoAVDzj1IzkdNLn8BMX5h0NynXMN8aN44mZ2Y+kVAGzLc7iGGwKT7m89NcizUmkNkVZnH7GM7//syZtBuToBOmYroKH4jrAXzMfiECMdJlfuKSf6gCUBwfyTXQEpToyvrxtPbeqWLwDTDiuwRhTdJKgNO+DVsZCM8W51jUAbAerrRRGwDy/hKcuxHw/lxmpo8MhRTBk4/rAfKwCSADyJAwZwhnjMq5h3PyhtxHi8FIEzAKjQgAAo3rASyeLbYDCsOQh00tANH+sRzZ9DOc+0ZpOV3C09PlAbzPJgogoSj/iAbQbDCIACYKwvBLAKneLz9lLoF2lSDUjjkj5JJh/2jDkrhNx7UM7km3l7YBxtswuZDkA3E9gEcTBgIdLSikLo/3OgAJl8BXy8poEmoJHb6/wOHYips+gb9H4nkAP+UOZAj7BgNhOPH8Qz5eiEA2uDzUXVxKHxQUUTgjO64HcAYwSPHBQTMwVU8OgNIysSPoVenjtA6Xyw0YS9odzt1zzZZzejyAgYjlImUWP61Y2cRqRV983VLA22xz0N6SMjprMFFEhwcklgF2x5tKAOjk/XpBcKq+PBWE9Pl2+13tducTOP8vADAQzwPktB/0D0BhP1EDIO9zP9wfA/k2surZcjO9WVJOPZIPJNUDuLWZvRUddvtDs0pKn9MKXv06s6Hd5voSgDw9z1p5tM1mD+v1j3anc9A/OFA1ANHXciY9hXL5qdkqXscgE86AZDYGucDmbFtgd/6i1Ww+qccDOHj2jvkOx6B/tGsAiD7mcxP2gBv2WwPKZZ7NFkB2LENAv8dNX4znAUogj+BpL9TpH6MmA2K1VkHI6HA6J8NQ1wHIfgTRH88DrvMPKe21AIzYA25Fm+/3mwDjgXaHYzO278JQ++N5gDjtRnbIIwzvc8akRAbEa7Ptdlu73dUFGDtgpqf1eAAHz2Uyw1SY+gCuG26dzjoAWQEP+VOr2XIp5T0gkbZIELIW2FzTACOICdkbADCQ8h6QSOtyu1uUq9SEVoOp2DBnmKJcpY7KecDNAvC58IDohqHzHhUAxwHAInxeGoZL37VVqqO7xWg8NO7af8u73W632y1++z8PlxBi5yygvwAAAABJRU5ErkJggg==';
```
Comment (vincss) [inline src/sleepingHelper.ts:34]: Keep a constant outside of the function.

File: src/sleepingSettings.ts [hunk 2]
```diff
@@ -24,6 +22,7 @@ export type Settings = {
     preventStop?: boolean,
     version?: string | false,
     favIcon?: string,
+    useServerIcon?: boolean,
```
Comment (vincss) [inline src/sleepingSettings.ts:25]: Maybe put a string to specify another favicon.

**Review by vincss — CHANGES_REQUESTED** [review]

File: src/sleepingHelper.ts (hunk 1 above)
Comment (markmetcalfe) [reply src/sleepingHelper.ts]: Done

File: src/sleepingSettings.ts (hunk 2 above)
Comment (markmetcalfe) [reply src/sleepingSettings.ts:26]: Done - added a new setting: `favIconPath` so you can specify a path other than `server-icon.png`

File: src/sleepingHelper.ts [hunk 3]
```diff
@@ -18,6 +21,33 @@ export const isPortTaken = (port: number) => new Promise<boolean>((resolve) => {
...
+
+export const getFavIcon = (settings: Settings) => {
+    if (settings.favIcon) {
+        return settings.favIcon;
+    }
+
+    let favIconPath = settings.favIconPath;
+    if (settings.useServerIcon) {
+        const minecraftDir = settings.minecraftWorkingDirectory ?? process.cwd();
+        favIconPath = `${minecraftDir}/server-icon.png`;
```
Comment (vincss) [inline src/sleepingHelper.ts:34]: you should use favIconPath here

Comment (markmetcalfe) [reply src/sleepingHelper.ts]: Done - have removed the `userServerIcon` setting

File: src/sleepingSettings.ts [hunk 4]
```diff
@@ -24,6 +22,8 @@ export type Settings = {
     preventStop?: boolean,
     version?: string | false,
     favIcon?: string,
+    favIconPath?: string,
```
[the remark this answers is not in the document]
Comment (markmetcalfe) [reply src/sleepingSettings.ts:25]: Done - have removed the `useServerIcon` parameter in favour of just having `favIconPath`

**Comment by markmetcalfe** [comment]
@vincss I've resolved the conflicts with the MOTD changes that were merged, and have made the changes you suggested. Once this is merged, could you run a release? I don't have any further contributions to make for now - I want to get a docker container working which I will need a release for.

File: src/sleepingHelper.ts [hunk 5]
```diff
@@ -22,6 +24,32 @@ export const isPortTaken = (port: number) => new Promise<boolean>((resolve) => {
...
+    if (settings.favIcon) {
+        return settings.favIcon;
+    }
+
+    let favIconPath = settings.favIconPath;
+    if (!favIconPath) {
+        return DefaultFavIconString;
+    }
+
+    if (!favIconPath.startsWith('/') && !favIconPath.startsWith('\\')) {
```
[the remark this answers is not in the document]
Comment (markmetcalfe) [reply src/sleepingHelper.ts]: Done

**Review by vincss — APPROVED** [review]

**Pull request merged**